### ISSUE TITLE
getQuerySelector(...)=>(state) expects 2 arguments

### ISSUE THREAD

**Issue by zeraien** [klis87/redux-requests#335]
getQuerySelector()=>() requires 2 arguments, but I think the second should be made optional.
This is not a syntax error, just a linting issue. Not sure I syntaxed that explanation correctly, so here is an example.
```
const getStuff = getQuerySelector({ type: types.LOAD_STUFF });

const stuff = getStuff(state);
```

In the above example, `getStuff(state)` will produce a linting error saying that the function requires 2 arguments, the second argument is expected to be `props`, the same stuff that's already been passed to the `getQuerySelector` call. 

Unless I misunderstand the way that the result of getQuerySelector should be used...
:)

**Comment by klis87** [comment]
You mean that Typescript types should be amended? Indeed 2nd prop should be removed as it is already passed by `getQuerySelector`

Wondering though where linter complains?

As I created `getQuerySelector` as convenience to pass to `reselect` or `useSelector` in React. If you want to use selector directly, it is easier just to:
`const stuff = getQuery(state, { type: types.LOAD_STUFF });`

**Comment by zeraien** [comment]
I like to prepare my selectors to use them in various react/redux connectors mainly, so I never use getQuery, I always set up getQuerySelector and use that around my code, for example:
```
const getObjectRequest = (object_id) => getQuerySelector({type:"OBJECT",requestKey:object_id})
```
Then I just use that in connect:
```
const object_request = getObjectRequest(ownProps.object_id);
```

This way, if I ever change the name of the action or the way the data is stored, I can just change the selector function.
In the above example, the linter complains here:`getObjectRequest(ownProps.object_id)`

The definition of `getQuerySelector` in `types/index.d.ts` seems to contain the second props argument...:

```typescript
export function getQuerySelector<QueryStateData = any>(props: {
  type: string;
  requestKey?: string;
  multiple?: boolean;
  defaultData?: any;
}): (
  state: any,
  props: {
    type: string;
    requestKey?: string;
    multiple?: boolean;
    defaultData?: any;
  },
) => QueryState<QueryStateData>;
```

![Screen Shot 2020-04-26 at 16 14 10](https://user-images.githubusercontent.com/134224/80305188-fcab6c80-87d8-11ea-9c62-bf835be378d1.png)
![Screen Shot 2020-04-26 at 16 13 02](https://user-images.githubusercontent.com/134224/80305178-ed2c2380-87d8-11ea-86b1-afb4497ab76c.png)

**Comment by zeraien** [comment]
Although I realise now I can just do...
```javascript
export const getWaypointGroups = (state) => getQuery(state,{type:types.LOAD_GROUPS})
```
;-)

**Comment by klis87** [comment]
@zeraien above example was specific reason why `getQuerySelector` was created, to save people one arrow function :)

I will fix this issue before the next release, which should come within 1 week. Don't be surprised though, it will actually have name change into `@redux-requests/core` and won't depend on saga anymore. Albeit usage with sagas will still be possible.

**Comment by klis87** [comment]
Fixed in `0.28.0`, thx for reporting this!
